src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -1,52 +1,128 @@
+
+    }
+
+    public GoGoDuckModule(String profile, IndexReader indexReader, String subset, GoGoDuckConfig goGoDuckConfig) {
         this.profile = profile;
         this.indexReader = indexReader;
         this.subset = new GoGoDuckSubsetParameters(subset);
-        this.userLog = userLog;
+        this.goGoDuckConfig = goGoDuckConfig;
     }
 
     public URIList getUriList() throws GoGoDuckException {
-        return indexReader.getUriList(profile, TIME_FIELD, URL_FIELD, subset);
+        return indexReader.getUriList(profile, goGoDuckConfig.getTimeField(), goGoDuckConfig.getFileUrlField(), subset);
+    }
+
+    public boolean hasPackedVariables() {
+        return containsPackedVariable;
     }
 
-    public void postProcess(File file) {
-        return;
+    public boolean isTimeUnlimited() {
+        return timeUnlimited;
     }
 
-    public List<String> ncksExtraParameters() {
-        List<String> ncksExtraParameters = new ArrayList<String>();
+    public void loadFileMetadata(File file) {
+        boolean containsPackedVariable = false;
+        GridDataset gridDs = null;
+
+        try {
+            setNcksSubsetParameters(file.getAbsoluteFile().toString());
+            setNcksExtraParameters();
+            this.timeUnlimited = time.isUnlimited();
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+            throw new GoGoDuckException(e.getMessage(), e);
+        }
+
+        try {
+            if (goGoDuckConfig.getUnpackNetcdf(profile)) {
+                // Checking the config (Needed for TPAC)
+                this.containsPackedVariable = true;
+            } else {
+                String location = file.getAbsolutePath();
+                Set<NetcdfDataset.Enhance> enhanceMode = new HashSet<>();
+                enhanceMode.add(NetcdfDataset.Enhance.ScaleMissing);
+                gridDs = GridDataset.open(location, enhanceMode);
+                for (Variable var : gridDs.getNetcdfDataset().getVariables()) {
+                    VariableDS vds = new VariableDS(var.getGroup(), var, true);
+                    if (vds.hasScaleOffset()) {
+                        containsPackedVariable = true;
+                        break;
+                    }
+                }
+                this.containsPackedVariable = containsPackedVariable;
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+            throw new GoGoDuckException(e.getMessage(), e);
+        } finally {
+            try {
+                if (gridDs != null) {
+                    gridDs.close();
+                }
+            } catch (IOException e) {
+                logger.error(e.getMessage());
+                throw new GoGoDuckException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private void setNcksExtraParameters() throws Exception {
+        ncksExtraParameters = new ArrayList<>();
+        for (String ncksParameter : goGoDuckConfig.getNcksParameters(profile)) {
```

Comment:
There are other parameters needed to be passed like "time,lat,lon,dt_analysis,l2p_flags,quality_level,satellite_zenith_angle,sea_surface_temperature,sses_bias,sses_count,sses_standard_deviation,sst_dtime", therefore I believe we still need config.